### ISSUE TITLE
Drone.io integration not working for repositories which use uppercase

### ISSUE THREAD

**Issue by 0xf3f** [gogits/gogs#2273]
Repositories are "advertised" to drone.io by their Name, which uses this name as a key in the DB. 
During creation of the outgoing webhook-payload, LowerName is used instead.

https://github.com/gogits/gogs/blob/6673dcb0380aa99da25b4d1d68cf129635fe30d9/models/repo.go#L405

The results is an 404 Error because the DB lookup failed.

![screenshot - 23 12 2015 - 17 54 15](https://cloud.githubusercontent.com/assets/7755130/11980865/3da48190-a99e-11e5-8674-63c3ab2fe875.png)

```
"repository": {
    "id": 4,
    "name": "test3",
    "url": "https://git.example.com/0xf3f/Test3",
```

**Comment by Unknwon** [comment]
Thanks your feedback!

But I think you should report this to https://github.com/drone/drone because code of integration is not written in Gogs.

**Comment by Unknwon** [comment]
cc @bradrydzewski

**Comment by 0xf3f** [comment]
But the payload of the webhook is provided by Gogs (as far as i understand it). The problem is, that the name of the repository is e.g. "Test" and not "test" as communicated in the name field.
Btw, he is aware of the issue:  https://github.com/drone/drone/issues/1357#issuecomment-160334143, but since no issue was filled, I thought I'll do it. Im not sure why 'LowerName' is used but its seams as an easy fix to me.

**Comment by Unknwon** [comment]
>  Im not sure why 'LowerName' is used but its seams as an easy fix to me.

Can you rephrase this sentence?

**Comment by 0xf3f** [comment]
```
func (repo *Repository) ComposePayload() *api.PayloadRepo {
	cl := repo.CloneLink()
	return &api.PayloadRepo{
		ID:          repo.ID,
		Name:        repo.LowerName,
		URL:         repo.FullRepoLink(),
		SSHURL:      cl.SSH,
		CloneURL:    cl.HTTPS,
		Description: repo.Description,
		Website:     repo.Website,
		Watchers:    repo.NumWatches,
		Owner: &api.PayloadAuthor{
			Name:     repo.MustOwner().DisplayName(),
			Email:    repo.MustOwner().Email,
			UserName: repo.MustOwner().Name,
		},
		Private: repo.IsPrivate,
	}
}
```
Am I right, that this function is user to create (parts of) the JSON payload. I assume that changing ``` Name:        repo.LowerName ``` to ```Name:        repo.Name``` will provide the right value.

**Comment by Unknwon** [comment]
Ah, I see.

**Comment by Unknwon** [comment]
Pushed fix to `develop`.

**Comment by Unknwon** [comment]
Close as fixed.
